src/tests/daos_racer.c

```diff
@@ -177,16 +184,47 @@ update_or_fetch(bool update)
 
 	for (i = 0; i < round; i++) {
 		int iod_nr = random % max_akey_per_dkey;
+		int cond_rand = rand();
+		uint64_t flags = 0;
 
 		memset(iods, 0, max_akey_per_dkey * sizeof(daos_iod_t));
 		pack_dkey_iod_sgl(dkey, &dkey_iov, akeys, iods, recxs, sgls,
 				  sgl_iovs, sgl_bufs, iod_nr);
-		if (update)
-			daos_obj_update(oh, DAOS_TX_NONE, 0, &dkey_iov, iod_nr,
-					iods, sgls, NULL);
-		else
-			daos_obj_fetch(oh, DAOS_TX_NONE, 0, &dkey_iov, iod_nr,
-				       iods, sgls, NULL, NULL);
+		if (update) {
+			if ((cond_rand % 100) < cond_pct) {
+				switch (cond_rand % 4) {
+				case 0:
+					flags = DAOS_COND_DKEY_INSERT;
+					break;
+				case 1:
+					flags = DAOS_COND_DKEY_UPDATE;
+					break;
+				case 2:
+					flags = DAOS_COND_AKEY_INSERT;
+					break;
+				case 3:
+					flags = DAOS_COND_AKEY_UPDATE;
+					break;
+				}
+			}
+
+			daos_obj_update(oh, DAOS_TX_NONE, flags, &dkey_iov,
+					iod_nr, iods, sgls, NULL);
```

Comment:
probably we should do cond_op and normal op on different dkey/akey or even different obj,  since the user will not do cond_op/op on the same object in the real world?

Reply:
That should be not important, because the server needs to handle the cases of condition unmatched. We more care about whether there is server corruption during the process.

Reply:
I mean matching the real world use cases, unless we do this in the real world as well?

Reply:
Our condition check is mainly based on dkey and akey, the name is generated randomly for each modification, so for current modification, it may exist, may not exist. If we want to increase the possibility of non-exist, then enlarge MAX_KEY_CNT.